### PR TITLE
api: simple status code extraction

### PR DESCRIPTION
**Pull request by jirikuncar**
- [x] extract meta information and merge them with `response` or simply commit event changes in on-going celery task

--
addresses #11

### CODE REVIEW

File: invenio_webhooks/models.py [hunk 1]
```diff
@@ -157,9 +157,28 @@ class CeleryReceiver(Receiver):
...
+
     def __call__(self, event):
         """Fire a celery task."""
-        process_event.apply_async(args=[event.id])
+        process_event.apply_async(task_id=str(event.id), args=[event.id])
+
+    def status(self, event):
+        """Return current processing status."""
+        result = AsyncResult(str(event.id))
+        return self.CELERY_STATES_TO_HTTP.get(result.state)
```
Comment (egabancho) [inline invenio_webhooks/models.py]: Could we think about returning also the `meta` by default?
Having custom states with progress seems to be quite a common use case (even documented) http://docs.celeryproject.org/en/latest/userguide/tasks.html#custom-states

Comment (jirikuncar) [inline invenio_webhooks/models.py]: I will do some more investigation in next PR where the `meta` will be included in HTTP headers.

**Comment by egabancho** [comment]
LGTM

File: tests/test_receivers.py [hunk 2]
```diff
@@ -165,3 +165,35 @@ def test_event_deletion(app, receiver):
...
+    """Test event status reporting."""
+    data = json.dumps({'x': 40, 'y': 2})
+
+    class TestCeleryReceiver(CeleryReceiver):
+
+        def run(self, event):
+            """Change task state."""
+            from celery import current_task, states
+
+            event.response['message'] = event.payload['x'] + event.payload['y']
```
Comment (omelkonian) [inline tests/test_receivers.py]: If I move the addition operation to a separate Celery task, I get this error:
```python
[2016-10-14 17:17:12,407] ERROR in models: Could not process event.
Traceback (most recent call last):
  File "/home/orestis/.virtualenvs/invenio-webhooks/src/invenio-webhooks/invenio_webhooks/models.py", line 268, in process
    self.receiver(self)
  File "/home/orestis/.virtualenvs/invenio-webhooks/src/invenio-webhooks/invenio_webhooks/models.py", line 171, in __call__
    process_event.apply_async(task_id=str(event.id), args=[event.id])
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/task.py", line 557, in apply_async
    link=link, link_error=link_error, **options)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/task.py", line 755, in apply
    request=request, propagate=throw)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/trace.py", line 355, in eager_trace_task
    uuid, args, kwargs, request)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/trace.py", line 253, in trace_task
    I, R, state, retval = on_error(task_request, exc, uuid)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/flask_celeryext/app.py", line 43, in __call__
    return Task.__call__(self, *args, **kwargs)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/task.py", line 420, in __call__
    return self.run(*args, **kwargs)
  File "/home/orestis/.virtualenvs/invenio-webhooks/src/invenio-webhooks/invenio_webhooks/models.py", line 149, in process_event
    db.session.add(event)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 482, in __exit__
    self._assert_active(deactive_ok=True, prepared_ok=True)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 223, in _assert_active
    raise sa_exc.ResourceClosedError(closed_msg)
ResourceClosedError: This transaction is closed
```

Therefore, I cannot run other tasks from the master task (i.e. the `CeleryReceiver::run` method).
Do you know why this happens?

Comment (omelkonian) [inline tests/test_receivers.py]: This is what I am doing, but I still get the error. The exact receiver code is this:
```python
    @shared_task()
    def add(x, y):
        return x + y

    class TestCeleryReceiver(CeleryReceiver):
        def run(self, event):            
            result = add.apply_async((event.payload['x'], event.payload['y']))
            event.response['message'] = result.get()
```
Should I begin a new session somewhere?

Comment (omelkonian) [inline tests/test_receivers.py]: Changing the `process_event` task session from `with db.session.begin_nested:` to `db.session.begin(nested=True, subtransactions=True)` seems to solve the problem.

Comment (omelkonian) [inline tests/test_receivers.py]: Also, just removing `with db.session.begin_nested():...` seems to work.

**Comment by jirikuncar** [comment]
Self-merging based on LGTM in https://github.com/inveniosoftware/invenio-webhooks/pull/38#issuecomment-253753339

**Pull request merged**